torch/nn/modules/module.py

```diff
@@ -1912,14 +1912,10 @@ def __setstate__(self, state):
         if "_backward_pre_hooks" not in self.__dict__:
             self._backward_pre_hooks = OrderedDict()
 
-    # On the return type:
-    # We choose to return `Any` in the `__getattr__` type signature instead of a more strict `Union[Tensor, Module]`.
-    # This is done for better interop with various type checkers for the end users.
-    # Having a stricter return type doesn't play nicely with `register_buffer()` and forces
-    # people to excessively use type-ignores, asserts, casts, etc.
-    # See full discussion on the problems with returning `Union` here
-    # https://github.com/microsoft/pyright/issues/4213
-    def __getattr__(self, name: str) -> Any:
+    # It is crucial that the return type is not annotated as `Any`, otherwise type checking
+    # on `torch.nn.Module` and all its subclasses is largely disabled as a result. See:
+    # https://github.com/pytorch/pytorch/pull/115074
+    def __getattr__(self, name: str) -> Union[Tensor, "Module"]:
```

Comment:
Please let's not add `Callable` to the return type.

That basically also means that we have almost turned off type checking on modules, because `Callable` just allows for **any** signature, and each of them return `Any`!

This means that ever `module.non_existing_function(...)` will just silently type check.